### PR TITLE
HIV-329  Allow upload of GeneXpert and DRT results in pdf and allow h…

### PR DESCRIPTION
**Pull request by madvincy**
…andle more than one page

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,12 +83,15 @@
...
+    "ngx-file-uploader": "0.0.14",
+    "ngx-formentry": "^2.8.0",
     "ngx-mydatepicker": "^2.4.9",
-    "ngx-openmrs-formentry": "git+https://github.com/AMPATH/ngx-openmrs-formentry.git#v2.12.2",
+    "ngx-openmrs-formentry": "^2.12.6",
     "ngx-pagination": "^3.1.1",
     "ngx-select": "^0.0.1",
     "ngx-toastr": "^9.1.1",
+    "ngx-webcam": "^0.2.5",
+    "npm": "^6.9.0",
```
Comment (nkimaina) [inline package.json]: Is this a Dev dependency or production code dependency? If Dev, move it to devDependencied:{}

File: src/app/patient-dashboard/common/forms/form-order-metadata.service.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export class FormOrderMetaDataService {
 
         return this.formsOrder;
     }
-    public setFavouriteForm(name: string) {
+    public setFavouriteForm(name: any) {
```
Comment (nkimaina) [inline src/app/patient-dashboard/common/forms/form-order-metadata.service.ts:23]: why switch to 'any'?

File: src/app/patient-dashboard/common/locator-map/locator-map.component.ts [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
 import { flatMap } from 'rxjs/operators';
 
+
 import * as _ from 'lodash';
 
 import { FileUploadResourceService } from '../../../etl-api/file-upload-resource.service';
 import { PersonResourceService } from '../../../openmrs-api/person-resource.service';
 import { AppSettingsService } from '../../../app-settings/app-settings.service';
 import { PatientService } from '../../services/patient.service';
+import { DataSource } from '@angular/cdk/table';
```
Comment (nkimaina) [inline src/app/patient-dashboard/common/locator-map/locator-map.component.ts]: 3rd party and angular imports usually go to the top of the file, before app imports. So it's usually
angular imports
..
3rd part imports
...
ampath and app imports

File: src/app/patient-dashboard/common/locator-map/locator-map.component.ts [hunk 4]
```diff
@@ -20,11 +22,17 @@ import { PatientService } from '../../services/patient.service';
...
   public loading = false;
+  public pdfAvailable = false;
   public imageSaved = false;
   public imageUploadFailed = false;
   public subscriptions = [];
+  public fileList = [];
+  public singleFileInput = true;
+  public openCamera = false;
   public patient: any;
+  innerValue = null;
```
Comment (nkimaina) [inline src/app/patient-dashboard/common/locator-map/locator-map.component.ts]: rename this variable appropriately

File: src/app/patient-dashboard/common/locator-map/locator-map.component.ts [hunk 5]
```diff
@@ -46,8 +54,24 @@ export class LocatorMapComponent implements OnInit, OnDestroy {
...
-  public onFileChange(file) {
+  public onFileChange(fileList) {
+    console.log(fileList.length);
+    for (const file of fileList) {
+      this.uploadFile(file);
+    }
+
+  }
+  public uploadFile(file) {
+    console.log(file);
```
Comment (nkimaina) [inline src/app/patient-dashboard/common/locator-map/locator-map.component.ts]: remove console.log

File: src/app/patient-dashboard/common/locator-map/locator-map.component.ts [hunk 6]
```diff
@@ -92,15 +116,22 @@ export class LocatorMapComponent implements OnInit, OnDestroy {
   public setPhoto() {
     const photo = this.patient.person.getPersonAttribute(this.attributeType);
     if (photo) {
+      console.log(photo);
```
Comment (nkimaina) [inline src/app/patient-dashboard/common/locator-map/locator-map.component.ts]: remove console.log

File: src/app/patient-dashboard/common/locator-map/locator-map.component.ts [hunk 7]
```diff
@@ -92,15 +116,22 @@ export class LocatorMapComponent implements OnInit, OnDestroy {
...
     } else {
       this.dataModel = null;
     }
 
   }
-
-  private getUrl() {
-    return this.appSettingsService.getEtlRestbaseurl().trim() + 'files/';
+  private setPdfUrl() {
+    this.fileUploadResourceService.getFile(this.dataModel, 'pdf').subscribe((file) => {
```
Comment (nkimaina) [inline src/app/patient-dashboard/common/locator-map/locator-map.component.ts:127]: we need a 'spinner' or a 'busy indicator' for this async call to the backend, just to show the user that something is happening. Add this to the various calls to the file upload resource service.

File: src/app/patient-dashboard/common/forms/form-order-metadata.service.ts (hunk 2 above)
Comment (madvincy) [reply src/app/patient-dashboard/common/forms/form-order-metadata.service.ts:23]: it's still a string but build breaks when declared as a string but passes when it's of type any

File: src/app/patient-dashboard/common/locator-map/locator-map.component.ts (hunk 7 above)
Comment (madvincy) [reply src/app/patient-dashboard/common/locator-map/locator-map.component.ts:127]: I have added a  spinner

**Review by nkimaina — DISMISSED** [review]

**Review by webtrendzs — APPROVED** [review]

**Review by nkimaina — APPROVED** [review]

**Pull request merged**